src/Expr.cc

```diff
@@ -3104,7 +3103,32 @@ Val* IndexExpr::Fold(Val* v1, Val* v2) const
 
 	switch ( v1->Type()->Tag() ) {
 	case TYPE_VECTOR:
-		v = v1->AsVectorVal()->Lookup(v2);
+		{
+		VectorVal* vect = v1->AsVectorVal();
+		const ListVal* lv = v2->AsListVal();
+
+		if ( lv->Length() == 1 )
+			v = vect->Lookup(v2);
+		else
+			{
+			int len = vect->Size();
+			VectorVal* result = new VectorVal(vect->Type()->AsVectorType());
+
+			bro_int_t first = get_slice_index(lv->Index(0)->CoerceToInt(), len);
```

Comment:
Agreed, let's make this a parse-time error.  Seems the consistent place to put that type-checking is in `VectorType::MatchesIndex`  -- e.g. probably just need to replicate the `IsIntegral` checks for both `el[0]` and `el[1]` list elements.

Reply:
I basically duplicated the string handling code below for getting the indexes which uses `ToInt()`. For some reason I whenever I passed a `count` value there it would throw an error with `ToInt()`, hence the coerce. I'll look into whether strings and vectors handle their index values differently. I agree that there should be some type checking there for integral types though.